core/src/main/java/org/jruby/util/RegularFileResource.java

```diff
@@ -218,7 +218,7 @@ public Channel openChannel(ModeFlags flags, int perm) throws ResourceException {
             return channel;
         }
 
-        if (file.isDirectory() && flags.isWritable()) {
+        if (file.isDirectory()) {
```

Comment:
@philr Should this happen on all platforms and not just windows?  Sorry this PR got lost in the cracks.

Reply:
@enebo On platforms other than Windows (with both MRI and JRuby), `File.open(directory)` succeeds and an `Errno::EISDIR` exception is raised when calling `File#read`.

On Windows, `File.open(directory)` raises an exception. MRI raises `Errno::EISDIR`, but JRuby raises `Errno::EACCES`.

To replicate the behaviour of MRI, it is just this code that handles Windows in the `openChannel()` method that needs to be changed.